tests/epyccel/test_epyccel_optional_args.py

```diff
@@ -0,0 +1,135 @@
+    ]
+)
+def test_f2(language):
+    @types('real')
+    def f2(x = None):
+        if x is None :
+            return 2.5
+        return x + 2.5
+
+    f = epyccel(f2, language = language)
+
+    # ...
+    assert f(2.0) == f2(2.0)
+    assert f() == f2()
+    assert f(None) == f2(None)
+    assert f(0.0) == f2(0.0)
+    # ...
+#------------------------------------------------------------------------------
+@pytest.mark.parametrize('language' , [
+    pytest.param("fortran", marks = [
+        pytest.mark.xfail(reason="f2py does not support optional arguments"),
+        pytest.mark.fortran]),
+    pytest.param("c" , marks = pytest.mark.c)
+])
+def test_f3(language):
+    @types('complex')
+    def f3(x = None):
+        if x is None :
+            return complex(2, 5.2)
+        return x + complex(2.5, 2)
+
+    f = epyccel(f3, language = language)
+
+    # ...
+    assert f(complex(1, 2.2)) == f3(complex(1, 2.2))
+    assert f() == f3()
+    assert f(None) == f3(None)
+    # ...
+#------------------------------------------------------------------------------
+@pytest.mark.parametrize( 'language', [
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="f2py does not support optional arguments"),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = pytest.mark.c)
+    ]
+)
+def test_f4(language):
+    @types('bool')
+    def f4(x = None):
+        if x is None :
+            return True
+        return False
+
+    f = epyccel(f4, language = language)
+
+    # ...
+    assert f(True) == f4(True)
+    assert f() == f4()
+    assert f(None) == f4(None)
+    assert f(False) == f4(False)
+    # ...
+#------------------------------------------------------------------------------
+@pytest.mark.parametrize( 'language', [
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="f2py does not support optional arguments"),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = pytest.mark.c)
+    ]
+)
+def test_f5(language):
+    import modules.Module_3 as mod
+
+    modnew = epyccel(mod, language = language)
+
+    # ...
+    assert mod.func(1) == modnew.func(1)
+    assert mod.func() == modnew.func()
+    assert mod.func(None) == modnew.func(None)
+    assert mod.func(0) == modnew.func(0)
+
+#------------------------------------------------------------------------------
+@pytest.mark.parametrize( 'language', [
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason="f2py does not support optional arguments"),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = pytest.mark.c)
+    ]
+)
+def test_f6(language):
+    import modules.Module_4 as mod
+
+    modnew = epyccel(mod, language = language)
+
+    # ...
+    assert mod.call_optional_1() == modnew.call_optional_1()
+    assert mod.call_optional_2(None) == modnew.call_optional_2(None)
+    assert mod.call_optional_2(0) == modnew.call_optional_2(0)

```

Comment:
Given that the function returns either `0` or `a`, you should provide something other than 0 as the value of a. Otherwise you don't know if the right loop was called or not

Reply:
Now this function returns `2` or `a`, hence this issue is resolved.